pydrad/parse/util.py

```diff
@@ -0,0 +1,228 @@
+"""
+Utilities related to parsing HYDRAD results
+"""
+import pathlib
+
+import astropy.table
+import numpy as np
+import plasmapy.particles
+from pandas import read_csv
+
+__all__ = [
+    'read_amr_file',
+    'read_phy_file',
+    'read_ine_file',
+    'read_trm_file',
+    'read_hstate_file',
+]
+
+
+def read_amr_file(filename):
+    """
+    Parse ``.amr`` files containing grid parameters and conserved quantities as
+    a function of position
+    """
+    # TODO: Account for possible presence of both electron
+    # and ion mass densities. If the electron mass density
+    # is present in this file, it will shift all of the
+    # columns from the index=2 position to the right.
+    columns = [
+        'grid_centers',
+        'grid_widths',
+        'ion_mass_density',
+        'momentum_density',
+        'electron_energy_density',
+        'ion_energy_density',
+    ]
+    units = {
+        'grid_centers': 'cm',
+        'grid_widths': 'cm',
+        'ion_mass_density': 'g cm-3',
+        'momentum_density': 'g cm-2 s-1',
+        'electron_energy_density': 'erg cm-3',
+        'ion_energy_density': 'erg cm-3',
+    }
+    table = astropy.table.QTable.read(
+        filename,
+        format='ascii',
+        data_start=4,
+    )
+    # NOTE: This is done after creating the table because the
+    # remaining number of columns can be variable and thus we
+    # cannot assign all of the column names at once.
+    table.rename_columns(
+        table.colnames[:len(columns)],
+        columns,
+    )
+    for column in columns:
+        table[column].unit = units[column]
+    return table
+
+
+def read_phy_file(filename):
+    """
+    Parse ``.phy`` files containing thermodynamic quantities as a function of position.
+    """
+    columns = [
+        'coordinate',
+        'velocity',
+        'sound_speed',
+        'electron_density',
+        'ion_density',
+        'electron_pressure',
+        'ion_pressure',
+        'electron_temperature',
+        'ion_temperature',
+        'electron_heat_flux',
+        'ion_heat_flux',
+    ]
+    units = {
+        'coordinate': 'cm',
+        'velocity': 'cm / s',
+        'sound_speed': 'cm / s',
+        'electron_density': 'cm-3',
+        'ion_density': 'cm-3',
+        'electron_pressure': 'dyne cm-2',
+        'ion_pressure': 'dyne cm-2',
+        'electron_temperature': 'K',
+        'ion_temperature': 'K',
+        'electron_heat_flux': 'erg s-1 cm-2',
+        'ion_heat_flux': 'erg s-1 cm-2',
+    }
+    return astropy.table.QTable.read(
+        filename,
+        format='ascii',
+        names=columns,
+        units=units,
+    )
+
+
+def read_ine_file(filename, n_s):
+    """
+    Parse ``.ine`` files containing ionization fraction as a function of position
```

Comment:
Similar to the comments above, I wonder if it's worth just clarifying what the file extensions mean.  For example, 
```suggestion
    Parse Ionization Non-Equilibrium ``.ine`` files containing ionization fraction as a function of position
```